### PR TITLE
fixed non-alpha header when first come up

### PR DESCRIPTION
**Pull request by neo5anderson**
the change diff like this:

```
  -<h2>\u87d2\u86c7</h2>
  +<h2 id="-1">\u87d2\u86c7</h2>
```

### CODE REVIEW

**Comment by neo5anderson** [comment]
the `header_id_from_text` generated headers-id not work for non-alpha characters when toc

sample input:

```
# Python

## 蟒蛇

- 外形特性
- 生活习性

## Markdown
## 标记语言

- 类型
- 历史
```

original test output:

```
 <ul>
  <li><a href="#python">Python</a>
  <ul>
    <li><a href="#markdown">Markdown</a></li>
    <li><a href="#-2">标记语言</a></li>
  </ul></li>
</ul>
```

but we want:

```
 <ul>
  <li><a href="#python">Python</a>
  <ul>
    <li><a href="#-1">蟒蛇</a></li>
    <li><a href="#markdown">Markdown</a></li>
    <li><a href="#-2">标记语言</a></li>
  </ul></li>
</ul>
```

File: test/test_markdown2.py [hunk 1]
```diff
@@ -166,7 +166,9 @@ def charreprreplace(exc):
                           _display(python_toc_html), _display(toc_html),
                           _indent(diff))
             self.assertEqual(python_norm_toc_html, norm_toc_html,
-                errmsg.encode('ascii', 'charreprreplace'))
+                errmsg.encode('utf-8', 'charreprreplace'))
+            # self.assertEqual(python_norm_toc_html, norm_toc_html,
+            #     errmsg.encode('ascii', 'charreprreplace'))
```
Comment (nicholasserra) [inline test/test_markdown2.py]: Looks good. Can you remove the commented out lines?

Comment (neo5anderson) [inline test/test_markdown2.py]: sure, but I'm not sure about whether the code need to be this, to CI and me, it works fine.

Comment (nicholasserra) [inline test/test_markdown2.py]: I think it's right. Should be utf-8. I'm not too worried as this is just the test code anyway.

**Comment by neo5anderson** [comment]
It removed

**Comment by nicholasserra** [comment]
Thanks :D

**Pull request merged**